index.js

```
@@ -237,74 +244,80 @@ function calcChecksumAsIndex( checksum, modulus ){
-	var article = removeHyphens(articleId);
+	var user         = removeHyphens(userId);
+	var shareDetails = user + timeString + tokensString + context;
+	validateStringOrThrow( 'shareDetails', shareDetails, detailsRegex );
 
-	var userTimeTokens = user + timeString + tokensString + context;
+	var sig = calcSigOfShareDetails( shareDetails, articleId, pem );
 
-	var userTimeTokensDictionaryIndexes  = dictionaryIndexes(userTimeTokens);
-	var checksumIndex 					 = calcChecksumAsIndex(calcIndiciesChecksum(userTimeTokensDictionaryIndexes), numPossibleChars);
-	userTimeTokensDictionaryIndexes.push( checksumIndex );
+	var shareCodeUnshuffled      = shareDetails + sig;
+	var shareCodeUnshuffledArray = shareCodeUnshuffled.split('');
+	var shareCodeArray           = seededShuffle(shareCodeUnshuffledArray, pem+articleId);
+	var shareCode                = shareCodeArray.join('');
+	validateStringOrThrow( 'shareCode', shareCode, codeRegex );
 
-	var shuffledUserTimeTokensDictionaryIndexes = seededShuffle(userTimeTokensDictionaryIndexes, salt + 'userTimeTokensDictionaryIndexes');
-
-	var articleDictionaryIndexes         = dictionaryIndexes(article);
-	var shuffledArticleDictionaryIndexes = seededShuffle(articleDictionaryIndexes, salt + 'articleDictionaryIndexes');
-	var saltDictionaryIndexes            = dictionaryIndexes(salt);
-
-	// ensure salt is always 2nd arg to addOverArrays
-	var tokenIndexes = addOverArrays(addOverArrays(shuffledUserTimeTokensDictionaryIndexes, shuffledArticleDictionaryIndexes), saltDictionaryIndexes)
-	.map(a => mod(a, numPossibleChars));
-
-	var code = dictionaryIndexesToString(tokenIndexes);
-
-	return code;
+	return shareCode;
 }
 
-function decrypt(code, article, salt) {
+function decrypt(code, articleId, pem) {
 
-	validateStringOrThrow(     'code',    code, codeRegex           );
-	validateStringOrThrow('articleId', article, uuidRegexWithHyphens);
-	validateStringOrThrow(     'salt',    salt, saltRegex           );
+	pem = pem || "";
```

yes, was kind of 'coding out loud'.

the call should fail if pem is not set, or is set incorrectly

that pem = pem step is not needed